python/pyspark/sql/connect/plan.py

```diff
@@ -1192,6 +1192,7 @@ def plan(self, session: "SparkConnectClient") -> proto.Relation:
         assert self._child is not None
 
         plan = proto.Relation()
+        plan.common.plan_id = self._child._plan_id
```

Comment:
Maybe we can add a comment here: we treat the id of the to-be-observed plan as the dataframe id for `CollectMetrics`

Reply:
I was following examples like
```
class CachedRelation(LogicalPlan):
    def __init__(self, plan: proto.Relation) -> None:
        super(CachedRelation, self).__init__(None)
        self._plan = plan
        # Update the plan ID based on the incremented counter.
        self._plan.common.plan_id = self._plan_id
```
So I assumed I need to manually set the plan_id somehow.

Reply:
I am doing this because `CollectMetrics` does not re-use `LogicalPlan`'s `_create_proto_relation` which helps set the `plan_id`. However I do not have context on why `CollectMetrcis` is implemented by the current way.

cc @zhengruifeng

Reply:
you change here is fine, it is equivalent to `_create_proto_relation`

I notice there are a few places not using `_create_proto_relation`, I will check and modify them in separate work